### PR TITLE
Add AWS SSO JARs to the `Standard.AWS` library

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

As reported by our users, when using the AWS SSO, our code was failing with:
```
Execution finished with an error: To use Sso related properties in the 'xyz' profile, the 'sso' service module must be on the class path.
```

This PR adds the missing JARs to fix that.

Additionally it improves the license review tool UX a bit (parts of #9122):
- sorting the report by amount of problems, so that dependencies with unresolved problems appear at the top,
- semi-automatic helper button to rename package configurations after a version bump,
- button to remove stale entries from config (files or copyrights that disappeared after update),
- button to add custom copyright notice text straight from the report UI,
- button to set a file as the license for the project (creating the `custom-license` file automatically)
- ability to filter processed projects - e.g. `openLegalReviewReport AWS` will only run on the AWS subproject - saving time processing unchanged dependencies,
- updated the license search heuristic, fixing a problem with duplicates:
	- if we had dependencies `netty-http` and `netty-http2`, because of a prefix-check logic, the notices for `netty-http` would also appear again for `netty-http2`, which is not valid. I have improved the heuristic to avoid these false positives and removed them from the current report.
- WIP: button to mark a license type as reviewed (not finished in this PR).

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
I think the JARs seem to have fixed the problem, I just now have to finish the license review.

I tried to create a simple repro (without access to a SSO account, a simplified repro):
I created a file `~/.aws/config` with contents:
```
[profile xyz]
sso_start_url = https://enso.org/
sso_region = us-east-1
sso_account_id = 1234
sso_role_name = role-name
region = eu-west-1
output = yaml
```

and tried running a script:
```
from Standard.Base import all
from Standard.AWS import all

main =
    cred = AWS_Credential.Profile "xyz"
    my_file = S3_File.new "s3://enso-test-bucket/foo.txt" credentials=cred
    IO.println my_file.read
```

Before this PR it failed, same as in the user report:
```
Execution finished with an error: To use Sso related properties in the 'xyz' profile, the 'sso' service module must be on the class path.
        at <java> software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.ssoCredentialsProviderFactory(ProfileCredentialsUtils.java:324)
        at <java> software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.ssoProfileCredentialsProvider(ProfileCredentialsUtils.java:191)
        at <java> software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.credentialsProvider(ProfileCredentialsUtils.java:120)
        at <java> software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.credentialsProvider(ProfileCredentialsUtils.java:102)
        at <java> software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider.lambda$createCredentialsProvider$1(ProfileCredentialsProvider.java:169)
        at <java> java.base/java.util.Optional.flatMap(Optional.java:289)
        at <java> software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider.createCredentialsProvider(ProfileCredentialsProvider.java:169)
        at <java> software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider.handleProfileFileReload(ProfileCredentialsProvider.java:135)
        at <java> software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider.resolveCredentials(ProfileCredentialsProvider.java:126)
        at <java> software.amazon.awssdk.core.internal.util.MetricUtils.measureDuration(MetricUtils.java:50)
        at <java> software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy.resolveCredentials(AwsCredentialsAuthorizationStrategy.java:100)
        at <java> software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy.addCredentialsToExecutionAttributes(AwsCredentialsAuthorizationStrategy.java:77)
        at <java> software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.invokeInterceptorsAndCreateExecutionContext(AwsExecutionContextBuilder.java:123)
        at <java> software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.invokeInterceptorsAndCreateExecutionContext(AwsSyncClientHandler.java:69)
        at <java> software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$0(BaseSyncClientHandler.java:64)
        at <java> software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:179)
        at <java> software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:62)
        at <java> software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:52)
        at <java> software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:63)
        at <java> software.amazon.awssdk.services.s3.DefaultS3Client.getObject(DefaultS3Client.java:4580)
        at <java> software.amazon.awssdk.services.s3.S3Client.getObject(S3Client.java:8077)
        at <enso> S3.get_object<arg-1>(X:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\AWS\0.0.0-dev\src\S3\S3.enso:137:16-39)
...
```

after the PR it is now failing to read some SSO cache - so the SSO logic seems to start, just cannot complete due to lack of full configuration.

```
Execution finished with an error: java.nio.file.NoSuchFileException: C:\Users\DELL\.aws\sso\cache\efc34417aa73a6f9c63a9fd8d90c404bb70cfd6b.json
        at <java> software.amazon.awssdk.services.sso.internal.SsoAccessTokenProvider.tokenFromFile(SsoAccessTokenProvider.java:58)
        at <java> software.amazon.awssdk.services.sso.internal.SsoAccessTokenProvider.resolveToken(SsoAccessTokenProvider.java:51)
        at <java> software.amazon.awssdk.services.sso.auth.SsoProfileCredentialsProviderFactory$SsoProfileCredentialsProvider.<init>(SsoProfileCredentialsProviderFactory.java:107)
        at <java> software.amazon.awssdk.services.s
...
```

I cannot guarantee this is enough to make sure that SSO works, but we surely moved forward.

**Comment by radeusgd** [comment]
Parts of the license review tool UX improvements are still WIP, but I wanted to move this PR forward so we can already provide the SSO support and test if it works in practice.

I will finish the remaining UX improvements in a follow-up PR.

File: build.sbt [hunk 1]
```diff
@@ -2966,8 +2968,11 @@ lazy val `std-aws` = project
       "com.amazonaws"          % "aws-java-sdk-core"       % awsJavaSdkV1Version,
       "com.amazonaws"          % "aws-java-sdk-redshift"   % awsJavaSdkV1Version,
       "com.amazonaws"          % "aws-java-sdk-sts"        % awsJavaSdkV1Version,
+      "software.amazon.awssdk" % "bom"                     % awsJavaSdkV2Version,
```
Comment (jdunkerley) [inline build.sbt]: alphabetical sort?

**Review by GregoryTravis — APPROVED** [review]

**Review by Akirathan — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]
Thanks Radek - great to have these improvements to legal review and have this feature for the user.

File: build.sbt (hunk 1 above)
Comment (radeusgd) [reply build.sbt]: can do

File: project/GatherLicenses.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import sbt.Keys._
-import sbt._
-import complete.DefaultParsers._
+import sbt.Keys.*
+import sbt.*
```
Comment (hubertp) [inline project/GatherLicenses.scala:2]: huh? That's not Scala. I'm a bit surprised this even passes the compiler

**Review by hubertp — APPROVED** [review]
Nit about `*`. Please revert that change.

File: project/GatherLicenses.scala (hunk 2 above)
Comment (radeusgd) [reply project/GatherLicenses.scala:2]: Hah, interesting, some automated tool did this for me - I'm not sure if this was `scalafmtSbt` or IntelliJ's formatter (likely the latter).

Sure, I will revert this change in my follow up PR.

> Just for fun, note that this **_is_** actually valid Scala. This is the [_preferred_ Scala 3 syntax](https://docs.scala-lang.org/scala3/reference/changed-features/imports.html) - the old `_` syntax is still supported in Scala 3 for compatibility purposes. What is surprising me is that our SBT is happily accepting Scala 3 syntax - I thought it still runs on Scala 2.

Comment (radeusgd) [reply project/GatherLicenses.scala:2]: Note that we already are using the `*` imports in `build.sbt` before that (not my changes):
image

Are you sure this needs changing? I guess we should then make this consistent.

**Pull request merged**